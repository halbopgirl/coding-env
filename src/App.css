.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button[type="submit"] {
  background: #007cf0;
  color: white;
  border: none;
  border-radius: 6px;
  padding: 0.7em 2em;
  font-size: 1.1em;
  font-weight: 600;
  cursor: pointer;
  margin-top: 1em;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.08);
  transition: background 0.2s, transform 0.1s;
}

button[type="submit"]:hover {
  background: #005bb5;
  transform: translateY(-2px) scale(1.03);
}

form>label {
  display: block;
  margin-bottom: 0.8em;
}

form input[type="number"] {
  margin-left: 0.5em;
  padding: 0.3em 0.7em;
  border-radius: 4px;
  border: 1px solid #ccc;
  font-size: 1em;
}

form {
  background: rgba(255, 255, 255, 0.07);
  padding: 2em 2.5em;
  border-radius: 12px;
  box-shadow: 0 4px 24px rgba(0, 0, 0, 0.07);
  display: inline-block;
  margin-bottom: 2em;
}

.App-header p {
  margin-top: 1.5em;
  font-size: 1.2em;
  font-weight: 500;
}
