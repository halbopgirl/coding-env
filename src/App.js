import './App.css';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import './i18n';

function sort(width, height, length, mass) {
  const volume = width * height * length;
  const isBulky = volume >= 1000000 || width >= 150 || height >= 150 || length >= 150;
  const isHeavy = mass >= 20;

  if (isBulky && isHeavy) return 'REJECTED';
  if (isBulky || isHeavy) return 'SPECIAL';
  return 'STANDARD';
}

function App() {
  const { t, i18n } = useTranslation();
  const [inputs, setInputs] = useState({ width: '', height: '', length: '', mass: '' });
  const [result, setResult] = useState('');

  const handleChange = (e) => {
    setInputs({ ...inputs, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const { width, height, length, mass } = inputs;
    const stack = sort(Number(width), Number(height), Number(length), Number(mass));
    setResult(stack);
  };

  return (
    <div className="App">
      <header className="App-header">
        <form onSubmit={handleSubmit}>
          <label>
            {t('width')}
            <input name="width" value={inputs.width} onChange={handleChange} type="number" required />
          </label>
          <br />
          <label>
            {t('height')}
            <input name="height" value={inputs.height} onChange={handleChange} type="number" required />
          </label>
          <br />
          <label>
            {t('length')}
            <input name="length" value={inputs.length} onChange={handleChange} type="number" required />
          </label>
          <br />
          <label>
            {t('mass')}
            <input name="mass" value={inputs.mass} onChange={handleChange} type="number" required />
          </label>
          <br />
          <button type="submit">{t('sort')}</button>
        </form>
        {result && (
          <p>{t('stack', { result: t(result) })}</p>
        )}
      </header>
    </div>
  );
}

export { sort };
export default App;
