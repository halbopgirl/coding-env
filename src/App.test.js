import { render, screen } from '@testing-library/react';
import App, { sort } from './App';

it('renders the app UI correctly (snapshot)', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

describe('sort', () => {
  it('returns REJECTED for heavy and bulky', () => {
    expect(sort(200, 10, 10, 25)).toBe('REJECTED'); // bulky (width) and heavy
    expect(sort(100, 100, 100, 25)).toBe('REJECTED'); // bulky (volume) and heavy
  });

  it('returns SPECIAL for only heavy', () => {
    expect(sort(10, 10, 10, 25)).toBe('SPECIAL');
  });

  it('returns SPECIAL for only bulky (dimension)', () => {
    expect(sort(151, 10, 10, 10)).toBe('SPECIAL');
  });

  it('returns SPECIAL for only bulky (volume)', () => {
    expect(sort(100, 100, 100, 10)).toBe('SPECIAL');
  });

  it('returns STANDARD for neither heavy nor bulky', () => {
    expect(sort(10, 10, 10, 10)).toBe('STANDARD');
  });
});
