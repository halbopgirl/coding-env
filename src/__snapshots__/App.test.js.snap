// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app UI correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <form>
        <label>
          Width (cm):
          <input
            name="width"
            required=""
            type="number"
            value=""
          />
        </label>
        <br />
        <label>
          Height (cm):
          <input
            name="height"
            required=""
            type="number"
            value=""
          />
        </label>
        <br />
        <label>
          Length (cm):
          <input
            name="length"
            required=""
            type="number"
            value=""
          />
        </label>
        <br />
        <label>
          Mass (kg):
          <input
            name="mass"
            required=""
            type="number"
            value=""
          />
        </label>
        <br />
        <button
          type="submit"
        >
          Sort Package
        </button>
      </form>
    </header>
  </div>
</DocumentFragment>
`;
